authzed/api/v1/permission_service.proto

```diff
@@ -250,6 +251,9 @@ message CheckPermissionRequest {
 
   // subject is the subject that will be checked for the permission or relation.
   SubjectReference subject = 4 [ (validate.rules).message.required = true ];
+
+  /** context consists of named values that are injected into the caveat evaluation context **/
+  google.protobuf.Struct context = 5 [ (validate.rules).message.required = false ];
 }
 
 message CheckPermissionResponse {
```

Comment:
Please add `PERMISSIONSHIP_MAYBE_PERMISSION` and a `repeated string missing_required_context = 3;`